### PR TITLE
forcescp can be cleared with --forcescp=false

### PR DESCRIPTION
**Pull request by gmarceau**
Addresses https://github.com/stellar/stellar-core/issues/421

### CODE REVIEW

File: src/main/main.cpp [hunk 1]
```diff
@@ -302,7 +313,7 @@ main(int argc, char* const* argv)
         Logging::setLogLevel(logLevel, nullptr);
 
         cfg.REBUILD_DB = newDB;
-        cfg.FORCE_SCP = forceSCP;
+        cfg.FORCE_SCP = (forceSCP != nullptr);
```
Comment (MonsieurNicolas) [inline src/main/main.cpp]: this is a break of contract: code depending on config.forceSCP expects it to be true iif forceSCP is set globally.

Comment (gmarceau) [inline src/main/main.cpp]: I'm not sure I see what you mean. `forceSCP` here is a local variable.

Comment (MonsieurNicolas) [inline src/main/main.cpp]: cfg.FORCE_SCP is used in the code with the value true when the db state is force_scp==true.

I guess this line is just bogus: cfg.FORCE_SCP should be left as "false" until it's forced to "true" when reading from the database

File: src/main/main.cpp [hunk 2]
```diff
@@ -219,7 +230,7 @@ main(int argc, char* const* argv)
     el::Level logLevel = el::Level::Info;
     std::vector<char*> rest;
 
-    bool forceSCP = false;
+    optional<bool> forceSCP = nullptr;
```
Comment (jedmccaleb) [inline src/main/main.cpp]: why this instead of a bool?

File: src/main/main.cpp (hunk 1 above)
Comment (jedmccaleb) [inline src/main/main.cpp]: yeah I agree we don't need to set cfg.FORCE_SCP here. The old code was also wrong

**Pull request merged**